main.go

```diff
@@ -24,10 +26,59 @@ const (
 
 // Configs ...
 type Configs struct {
-	ProjectLocation string `env:"project_location,required"`
+	ProjectLocation string `env:"project_location,dir"`
 	APKPathPattern  string `env:"apk_path_pattern"`
 	Variant         string `env:"variant"`
 	Module          string `env:"module"`
+	Arguments       string `env:"arguments"`
+	CacheLevel      string `env:"cache_level,opt[none,only_deps,all]"`
+}
+
+func getArtifacts(gradleProject gradle.Project, started time.Time, pattern string, includeModule bool) (artifacts []gradle.Artifact, err error) {
+	artifacts, err = gradleProject.FindArtifacts(started, pattern, includeModule)
+	if err != nil {
+		return
+	}
+	if len(artifacts) == 0 {
+		if !started.IsZero() {
+			log.Warnf("No artifacts found with pattern: %s that has modification time after: %s", pattern, started)
+			log.Warnf("Retrying without modtime check....")
+			fmt.Println()
+			return getArtifacts(gradleProject, time.Time{}, pattern, includeModule)
+		}
+		log.Warnf("No artifacts found with pattern: %s without modtime check", pattern)
+		log.Warnf("If you have changed default report export path in your gradle files then you might need to change ReportPathPattern accordingly.")
+	}
+	return
+}
+
+func exportArtifacts(artifacts []gradle.Artifact, deployDir string) ([]string, error) {
+	var exportedArtifactPaths []string
+	for _, artifact := range artifacts {
+		exists, err := pathutil.IsPathExists(filepath.Join(deployDir, artifact.Name))
+		if err != nil {
+			return nil, fmt.Errorf("failed to check path, error: %v", err)
+		}
+
+		artifactName := filepath.Base(artifact.Path)
+
+		if exists {
+			timestamp := time.Now().Format("20060102150405")
+			ext := filepath.Ext(artifact.Name)
+			name := strings.TrimSuffix(filepath.Base(artifact.Name), ext)
+			artifact.Name = fmt.Sprintf("%s-%s%s", name, timestamp, ext)
```

Comment:
why? this block would be more clear. if you would just use the artifact.Name. Use a new variable to build the export artifact name.